toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+func enableVerityPartition(Verity imagecustomizerapi.Verity, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	// Ensure the VerityPartition value is the supported type: 'root'. TODO: add another supported type 'user'
+	if Verity.VerityTab != "root" {
+		return fmt.Errorf("invalid VerityPartition: %s. It should be 'root' only at this time.", Verity.VerityTab)
+	}
+
+	// Integrate systemd veritysetup dracut module into initramfs img.
+	systemdVerityDracutModule := "systemd-veritysetup"
+	err = buildDracutModule(systemdVerityDracutModule, imageChroot) 
+	if err != nil {
+		return err
+	}
+
+	// Update mariner config file with the new generated initramfs file.
+	err = updateMarinerCfgWithInitramfs(imageChroot)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildDracutModule(dracutModuleName string, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	// This function will be run inside the chroot to list kernel files in /boot
+	listKernels := func() ([]string, error) {
+		var kernels []string
+		// Assuming the vmlinuz files are located in /boot
+		files, err := filepath.Glob("/boot/vmlinuz-*")
+		if err != nil {
+			return nil, err
+		}
+		for _, file := range files {
+			kernels = append(kernels, filepath.Base(file))
+		}
+		return kernels, nil
+	}
+
+	var kernelFiles []string
+	err = imageChroot.Run(func() error {
+		kernelFiles, err = listKernels()
+		return err
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list kernels in chroot: %w", err)
+	}
+
+	if len(kernelFiles) == 0 {
+		return fmt.Errorf("no kernels found in chroot environment")
+	}
+
+	// Extract the version from the kernel filename (e.g., vmlinuz-5.15.131.1-2.cm2 -> 5.15.131.1-2.cm2)
+	kernelVersion := strings.TrimPrefix(kernelFiles[0], "vmlinuz-")
+
+	buildDracutModuleArgs := []string{
+		"dracut", "-f", "--kver", kernelVersion, "-a",
```

Comment:
Will address [comment](https://github.com/microsoft/CBL-Mariner/pull/6679) from [cwize1](https://github.com/cwize1).

Reply:
The `systemd-veritysetup` module exists by default in `/etc/dracut.conf.d/`. However, it will not be included automatically in the dracut install/build process because no dependencies trigger the inclusion of this module. Therefore, I need to include it manually by using the `-a` flag and specifying the name of this module.

Reply:
In the [dracut config](https://man7.org/linux/man-pages/man5/dracut.conf.5.html), there is the `add_dracutmodules` command.